lib/services/local/headlessauthn_watcher.go

```diff
@@ -219,33 +223,34 @@ func (h *HeadlessAuthenticationWatcher) notify(headlessAuthns ...*types.Headless
 	}
 }
 
+type WaitConditionFunc func(*types.HeadlessAuthentication) (bool, error)
+
 // Wait watches for the headless authentication with the given id to be added/updated
 // in the backend, and waits for the given condition to be met, to result in an error,
 // or for the given context to close.
-func (h *HeadlessAuthenticationWatcher) Wait(ctx context.Context, name string, cond func(*types.HeadlessAuthentication) (bool, error)) (*types.HeadlessAuthentication, error) {
+func (h *HeadlessAuthenticationWatcher) Wait(ctx context.Context, name string, cond WaitConditionFunc) (*types.HeadlessAuthentication, error) {
+	var initialBackendCheck = true
 	checkBackend := func() (*types.HeadlessAuthentication, bool, error) {
+		defer func() { initialBackendCheck = false }()
 		headlessAuthn, err := h.identityService.GetHeadlessAuthentication(ctx, name)
 		if err != nil {
+			if trace.IsNotFound(err) && initialBackendCheck {
+				// Ignore not found errors in the initial backend check.
+				return nil, false, nil
+			}
 			return nil, false, trace.Wrap(err)
 		}
-
 		ok, err := cond(headlessAuthn)
-		if err != nil {
-			return nil, false, trace.Wrap(err)
+		if err == nil && ok {
+			return headlessAuthn, true, nil
 		}
-
-		return headlessAuthn, ok, nil
+		return nil, false, trace.Wrap(err)
```

Comment:
Maybe:

```go
		return headlessAuthn, ok, trace.Wrap(err)
```

It's not exactly the same, but it's likely alright (and arguably easier to read).

Reply:
The only issue is that we can have `ok==false` and `headlessAuthn != nil`. Is this alright conventionally?